class.jetpack.php

```diff
@@ -6933,21 +6933,24 @@ public static function build_raw_urls( $url ) {
 	public static function dns_prefetch( $new_urls = null ) {
 		static $prefetch_urls = array();
 		if ( empty( $new_urls ) && ! empty( $prefetch_urls ) ) {
-			echo "\r\n";
-			foreach ( $prefetch_urls as $this_prefetch_url ) {
-				printf( "<link rel='dns-prefetch' href='%s'/>\r\n", esc_attr( $this_prefetch_url ) );
+			if( 'dns-prefetch' == $relation_type ) {
+				foreach ( (array) $prefetch_urls as $this_prefetch_url ) {
+					$urls[] = strtolower( untrailingslashit( preg_replace( '#^https?://#i', '//', $this_prefetch_url ) ) );
+				}
+				$urls = array_unique( $urls );
+			return $urls;
 			}
 		} elseif ( ! empty( $new_urls ) ) {
-			if ( ! has_action( 'wp_head', array( __CLASS__, __FUNCTION__ ) ) ) {
-				add_action( 'wp_head', array( __CLASS__, __FUNCTION__ ) );
-			}
-			foreach ( (array) $new_urls as $this_new_url ) {
-				$prefetch_urls[] = strtolower( untrailingslashit( preg_replace( '#^https?://#i', '//', $this_new_url ) ) );
-			}
-			$prefetch_urls = array_unique( $prefetch_urls );
+			if( 'dns-prefetch' == $relation_type )
+				foreach ( (array) $new_urls as $this_new_url ) {
+					$urls[] = strtolower( untrailingslashit( preg_replace( '#^https?://#i', '//', $this_new_url ) ) );
+				}
+				$urls = array_unique( $urls );
+			return $urls;
 		}
 	}
-
+	add_filter( 'wp_resource_hints', 'dns_prefetch', 10, 2 );
```

Comment:
This can't be placed here (within a class but not within a function). You'll see that other filters are added in the constructor: https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L508

Also, the filter needs to use the right callable. `dns_prefetch` would just look for a regular function by that name, not a method function within a class. https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L541 is an example of how to ask the hook to use a class method.

Reply:
Removed in new commit and I will push into `class.jetpack.php` similarly to the line you've mentioned

Reply:
OK, line now moved :)